_posts/2022-01-10-reassign-partitions.md

```diff
@@ -0,0 +1,416 @@
+        Topic: my-topic-two     Partition: 4    Leader: 3       Replicas: 3,2,0 Isr: 3,2,0
+        Topic: my-topic-two     Partition: 5    Leader: 0       Replicas: 0,1,4 Isr: 0,1,4
+        Topic: my-topic-two     Partition: 6    Leader: 1       Replicas: 1,4,2 Isr: 1,4,2
+        Topic: my-topic-two     Partition: 7    Leader: 4       Replicas: 4,2,3 Isr: 4,2,3
+        Topic: my-topic-two     Partition: 8    Leader: 2       Replicas: 2,3,0 Isr: 2,3,0
+        Topic: my-topic-two     Partition: 9    Leader: 3       Replicas: 3,0,1 Isr: 3,0,1
+```
+In the same way you can get the details for the other topic`my-topic` also.
+
+```sh
+Topic: my-topic TopicId: WyFKVZzLS8i54IGgm1ifrQ PartitionCount: 10      ReplicationFactor: 3    Configs: min.insync.replicas=2,segment.bytes=1073741824,retention.ms=7200000,message.format.version=3.0-IV1
+        Topic: my-topic Partition: 0    Leader: 0       Replicas: 0,2,1 Isr: 0,2,1
+        Topic: my-topic Partition: 1    Leader: 2       Replicas: 2,1,0 Isr: 0,2,1
+        Topic: my-topic Partition: 2    Leader: 1       Replicas: 1,0,2 Isr: 0,2,1
+        Topic: my-topic Partition: 3    Leader: 0       Replicas: 0,1,2 Isr: 0,2,1
+        Topic: my-topic Partition: 4    Leader: 2       Replicas: 2,0,1 Isr: 0,2,1
+        Topic: my-topic Partition: 5    Leader: 1       Replicas: 1,2,0 Isr: 0,2,1
+        Topic: my-topic Partition: 6    Leader: 0       Replicas: 0,2,1 Isr: 0,2,1
+        Topic: my-topic Partition: 7    Leader: 2       Replicas: 2,1,0 Isr: 0,2,1
+        Topic: my-topic Partition: 8    Leader: 1       Replicas: 1,0,2 Isr: 0,2,1
+        Topic: my-topic Partition: 9    Leader: 0       Replicas: 0,1,2 Isr: 0,2,1
+```
+
+From the above outputs, we got to know that my-topic-two has some replicas on broker 3 and 4 thus we need to reassign this topic onto other brokers. 
+
+Let's create our `topics.json` file:
+```json
+{
+  "version": 1,
+  "topics": [
+    { "topic": "my-topic-two"}
+  ]
+}
+```
+
+After creating this file, copy it to the interactive pod container since we will be running all of our commands from there:
+
+```sh
+kubectl cp topics.json <INTERACTIVE-POD-NAME>:/tmp/topics.json
+```
+
+We can now start a shell process in our interactive pod container and run the command inside it to generate our proposal `reassignment.json` data.
+Let's start the shell process:
+
+```sh
+kubectl exec -ti <INTERACTIVE-POD-NAME> /bin/bash
+```
+
+Now it's time to generate the `reassignment.json` file. 
+We will use the `kafka-reassign-partitions.sh` command to generate our proposal `reassignment.json` data:
+
+```sh
+bin/kafka-reassign-partitions.sh --bootstrap-server <CLUSTER-NAME>-kafka-bootstrap:9092 \
+--topics-to-move-json-file /tmp/topics.json \
+--broker-list 0,1,2 \
+--generate
+```
+Here `topics-to-move-json-file` points towards the `topics.json file` and `--broker-list` is the list of broker which will now handle the partitions which are present with the broker to be scaled down.
```

Comment:
`broker` -> `brokers`
`which will now handle the partitions which are present with the broker to be scaled down.` -> `we want to move our partitions onto`